shell/deploy-multi-chain.sh

```diff
@@ -0,0 +1,232 @@
+chains["sepolia"]="$SEPOLIA_RPC_URL $ETHERSCAN_API_KEY $SEPOLIA_CHAIN_ID $SEPOLIA_ADMIN $SEPOLIA_COMPTROLLER"
+chains["scroll"]="$SCROLL_RPC_URL $SCROLLSCAN_API_KEY $SCROLL_CHAIN_ID $SCROLL_ADMIN $SCROLL_COMPTROLLER"
+
+# Flag for broadcast deployment
+BROADCAST_DEPLOYMENT=false
+
+# Flag for deterministic deployment
+DETERMINISTIC_DEPLOYMENT=false
+
+# Flag for gas price
+GAS_PRICE=0
+WITH_GAS_PRICE=false
+
+# Flag for all chains
+ON_ALL_CHAINS=false
+
+# Provided chains
+provided_chains=()
+
+# Check for arguments passed to the script
+for ((i=1; i<=$#; i++)); do
+    arg=${!i}
+
+    # Check if '--broadcast' flag is provided the arguments
+    if [[ $arg == "--broadcast" ]]; then
+        BROADCAST_DEPLOYMENT=true
+    fi
+
+    # Check if '--deterministic' flag is provided in the arguments
+    if [[ $arg == "--deterministic" ]]; then
+        DETERMINISTIC_DEPLOYMENT=true
+    fi
+
+    # Check if '--with-gas-price' flag is provided in the arguments
+    if [[ $arg == "--with-gas-price" ]]; then
+        WITH_GAS_PRICE=true
+
+        # Increment index to get the next argument, which should be the gas price
+        ((i++))
+        GAS_PRICE=${!i}
+        if ! [[ $GAS_PRICE =~ ^[0-9]+$ ]]; then
+            echo -e "${EC}Error: Invalid value for --with-gas-price, must be number${NC}"
+            exit 1
+        fi
+    fi
+
+    # Check if '--all' flag is provided in the arguments
+    if [[ $arg == "--all" ]]; then
+        ON_ALL_CHAINS=true
+        provided_chains=("${!chains[@]}")
+    fi
+
+    # Check for passed chains
+    if [[ $arg != "--all" && $arg != "--deterministic" && $arg != "--broadcast" && $arg != "--with-gas-price" && $ON_ALL_CHAINS == false ]]; then
+        provided_chains+=("$arg")
+    fi
+done
+
+# Set the default chain to Sepolia if no chains are provided
+if [ ${#provided_chains[@]} -eq 0 ]; then
+    provided_chains=("sepolia")
+fi
+
+# Compile the contracts
+echo "Compiling the contracts..."
+FOUNDRY_PROFILE=optimized forge build
+
+# Deploy to the provided chains
+for chain in "${provided_chains[@]}"; do
+    # Check if the provided chain is defined
+    if [[ ! -v "chains[$chain]" ]]; then
+        echo -e "\n${WC}Warning: Chain configuration for '$chain' not found.${NC}"
+        continue
+    fi
+
+    # Split the configuration into RPC, API key, Chain ID, admin, and comptroller
+    IFS=' ' read -r rpc_url api_key chain_id admin comptroller <<< "${chains[$chain]}"
+
+    # Declare a deployment command
+    deployment_command="";
+
+    # Choose the script based on the flag
+    if [[ $DETERMINISTIC_DEPLOYMENT == true ]]; then
+        echo -e "\n${IC}Deploying deterministic contracts to $chain...${NC}"
+        # Construct the command
+        deployment_command="forge script script/DeployDeterministicCore3.s.sol \
+        --rpc-url $rpc_url \
+        --sig run(string,address,address,uint256) \
+        \'ChainID $chain_id, Version 1.1.1\' \
```

Comment:
This is not correct. It generates '\'\''ChainID' 11155111, Version '1.1.0\'\''' and now 'ChainID $chain_id, Version 1.1.1'

I've fixed it here https://github.com/sablier-labs/v2-core/pull/753

Reply:
gotcha, feel free to push directly @smol-ninja.